content/id/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -0,0 +1,221 @@
+
+{{% /capture %}}
+
+{{% capture body %}}
+## Custom resources
+
+Sebuah *resource* adalah sebuah *endpoint* pada [Kubernetes API](/docs/reference/using-api/api-overview/) yang menyimpan sebuah koleksi dari [objek API](/docs/concepts/overview/working-with-objects/kubernetes-objects/) dari sebuah jenis tertentu. Sebagai contoh, *resource* bawaan Pod mengandung sebuah koleksi objek-objek Pod.
+
+Sebuah Custom Resource adalah sebuah ekstensi dari Kubernetes API yang tidak seharusnya tersedia pada pemasangan default Kubernetes. Namun, banyak fungsi-fungsi inti Kubernetes yang sekarang dibangun menggunakan Custom Resource, membuat Kubernetes lebih modular.
+
+Custom Resources bisa muncul dan menghilang dalam sebuah klaster yang berjalan melalui *dynamic registration*, dan admin-admin klaster bisa memperbaharui Custom Resources secara independen dari klaster itu sendiri. Ketika sebuah Custom Resource
+dipasang, pengguna dapat membuat dan mengakses objek-objek Custom Resource menggunakan [kubectl](/docs/user-guide/kubectl-overview/), seperti yang mereka lakukan untuk *resource* bawaan seperti Pod.
+
+## Custom controllers
+
+Dengan sendirinya, *custom resources* memungkinkan kamu untuk menyimpan dan mengambil data terstruktur. Ketika kamu menggabungkan sebuah Custom Resource dengan *custom controller*, Custom Resource akan memberikan sebuah API deklaratif yang sebenarnya.
+
+Sebuah [API deklaratif](/docs/concepts/overview/working-with-objects/kubernetes-objects/#understanding-kubernetes-objects)
+memungkinkan kamu untuk mendeklarasikan atau menspesifikasikan keadaan dari *resource* kamu dan mencoba untuk menjaga keadaan saat itu tersinkronisasi dengan keadaan yang diinginkan. *Controller* menginterpretasikan data terstruktur sebagai sebuah rekaman dari keadaan yang diinginkan pengguna, dan secara kontinu menjaga keadaan ini.
+
+Kamu bisa men-*deploy* dan memperbaharui sebuah *custom controller* pada sebuah klaster yang berjalan, secara independen dari siklus hidup klaster itu sendiri. *Custom controller* dapat berfungsi dengan *resource* jenis apapun, tetapi mereka sangat efektif ketika dikombinasikan dengan Custom Resource. [Operator pattern](https://coreos.com/blog/introducing-operators.html) mengkombinasikan Custom Resource dan *custom controller*. Kamu bisa menggunakan *custom controller* untuk menyandi pengetahuan domain untuk aplikasi spesifik menjadi sebuah ekstensi dari Kubernetes API.
+
+## Haruskah saya menambahkan Custom Resource ke dalam klaster Kubernetes saya?
+
+Ketika membuat sebuah API baru, pikirkan apakah kamu ingin [mengagregasikan API kamu dengan API klaster Kubernetes](/docs/concepts/api-extension/apiserver-aggregation/) atau membiarkan API kamu berdiri sendiri.
+
+| Pilih agregasi API jika: | Pilih sebuah API yang berdiri sendiri jika: |
+| ---------------------------- | ---------------------------- |
+| API kamu bersifat [Deklaratif](#declarative-apis). | API kamu tidak cocok dengan model [Deklaratif](#declarative-apis). |
```

Comment:
The anchor has changed because of translation
```suggestion
| API kamu bersifat [Deklaratif](#api-deklaratif). | API kamu tidak cocok dengan model [Deklaratif](#api-deklaratif). |
```